ogr/ogrsf_frmts/hana/ogrhanatablelayer.cpp

```diff
@@ -0,0 +1,1762 @@
+            }
+            rsIdentity->close();
+        }
+
+        return err;
+    }
+    catch (const std::exception& ex)
+    {
+        CPLError( CE_Failure, CPLE_NotSupported,
+                  "Unable to create feature: %s", ex.what());
+        return OGRERR_FAILURE;
+    }
+    return OGRERR_NONE;
+}
+
+/************************************************************************/
+/*                           DeleteFeature()                            */
+/************************************************************************/
+
+OGRErr OGRHanaTableLayer::DeleteFeature(GIntBig nFID)
+{
+    if (!updateMode_)
+    {
+        CPLError(
+            CE_Failure, CPLE_NotSupported, UNSUPPORTED_OP_READ_ONLY,
+            "DeleteFeature");
+        return OGRERR_FAILURE;
+    }
+
+    if (nFID == OGRNullFID)
+    {
+        CPLError(
+            CE_Failure, CPLE_AppDefined,
+            "DeleteFeature(" CPL_FRMT_GIB
+            ") failed.  Unable to delete features "
+            "in tables without\n a recognised FID column.",
+            nFID);
+        return OGRERR_FAILURE;
+    }
+
+    if (OGRNullFID == fidFieldIndex_)
+    {
+        CPLError(
+            CE_Failure, CPLE_AppDefined,
+            "DeleteFeature(" CPL_FRMT_GIB
+            ") failed.  Unable to delete features "
+            "in tables without\n a recognised FID column.",
+            nFID);
+        return OGRERR_FAILURE;
+    }
+
+    if (deleteFeatureStmt_.isNull())
+    {
+        deleteFeatureStmt_ = CreateDeleteFeatureStatement();
+        if (deleteFeatureStmt_.isNull())
+            return OGRERR_FAILURE;
+    }
+
+    deleteFeatureStmt_->setLong(1, odbc::Long(static_cast<std::int64_t>(nFID)));
+    bool withBatch = dataSource_->IsTransactionStarted();
+    if (withBatch)
+       deleteFeatureStmt_->addBatch();
+
+    auto ret = ExecuteUpdate(*deleteFeatureStmt_, withBatch, "DeleteFeature");
+    return (OGRERR_NONE == ret.first && ret.second != 1)
+               ? OGRERR_NON_EXISTING_FEATURE
+               : ret.first;
+}
+
+/************************************************************************/
+/*                             ISetFeature()                            */
+/************************************************************************/
+
+OGRErr OGRHanaTableLayer::ISetFeature(OGRFeature* feature)
+{
+    if (!updateMode_)
+    {
+        CPLError(
+            CE_Failure, CPLE_NotSupported, UNSUPPORTED_OP_READ_ONLY,
+            "SetFeature");
+        return OGRERR_FAILURE;
+    }
+
+    if( nullptr == feature )
+    {
+        CPLError( CE_Failure, CPLE_AppDefined,
+                  "NULL pointer to OGRFeature passed to SetFeature()." );
+        return OGRERR_FAILURE;
+    }
+
+    if (feature->GetFID() == OGRNullFID)
+    {
+        CPLError(
+            CE_Failure, CPLE_AppDefined,
+            "FID required on features given to SetFeature().");
+        return OGRERR_FAILURE;
+    }
+
+    if (nullptr == feature)
```

Comment:
already tested above

Reply:
Removed in f2f6378e7df638dcb28936709e56af9c1c25127d.